NetworkLayer/Model/Services/AdService.swift

```diff
@@ -0,0 +1,40 @@
+//
+//  Service.swift
+//  NetworkLayer
+//
+//  Created by Nicolas Demange on 15/12/2022.
+//
+
+import Foundation
+
+// MARK: - Protocol
+
+protocol AdService {
+    
+    func fetchAdWithAsync() async throws
+    
+}
+
+// MARK: - Class
+
+final class AdSession: AdService {
+
+    // MARK: - Properties
+
+    
+    // MARK: - Initializer
+    
+    
+    // MARK: - Methods
+    
+    func fetchAdWithAsync() async throws {
+        guard let url = URL(string: "https://raw.githubusercontent.com/leboncoin/paperclip/master/listing.json") else  { fatalError("Missing URL") }
```

Comment:
```suggestion
        guard let url = URL(string: "https://raw.githubusercontent.com/leboncoin/paperclip/master/listing.json") else  { throw uneErreur }
```
Attention au fatalError. Throw une custom erreur à la place.